platforms/ide/problems-api/src/main/java/org/gradle/api/problems/Problem.java

```diff
@@ -18,86 +18,11 @@
 
 import org.gradle.api.Incubating;
 
-import javax.annotation.Nullable;
-import java.util.List;
-
 /**
  * Interface for describing structured information about a problem.
  *
  * @since 8.13
  */
 @Incubating
 public interface Problem {
```

Comment:
we moved everything from the public interface to `InternalProblem` because there is no need for the details to be publicly available.

This caused a lot of changes internally to use the `InternalProblem` instead of `Problem`